services/ui-src/src/types/index.ts

```diff
@@ -138,7 +138,7 @@ export interface ReportKeys {
   id: string;
 }
 
-export interface ReportShape extends ReportKeys {
+export interface DashboardReportShape extends ReportKeys {
```

Comment:
Creating an interface called DashboardReportShape and having ReportShape extend off of it made things a LOT easier. It could be argued that reportShape could just have formTemplate and fieldData swapped to an optional param but going that route includes a lot more rewrites, type checks, and testing.

Reply:
This makes a lot of sense to me, but I would suggest naming it something more descriptive of what it actually is, rather than where it's used, maybe like `ReportMetadataShape`.